dns.go

```diff
@@ -231,22 +224,6 @@ func (api *API) UpdateDNSRecord(ctx context.Context, rc *ResourceContainer, para
 
 	params.Name = toUTS46ASCII(params.Name)
 
-	// Populate the record name from the existing one if the update didn't
-	// specify it.
-	if params.Name == "" || params.Type == "" {
-		rec, err := api.GetDNSRecord(ctx, rc, params.ID)
-		if err != nil {
-			return err
-		}
-
-		if params.Name == "" {
-			params.Name = rec.Name
-		}
-		if params.Type == "" {
-			params.Type = rec.Type
-		}
-	}
-
 	uri := fmt.Sprintf("/zones/%s/dns_records/%s", rc.Identifier, params.ID)
```

Comment:
we should add a check above this to ensure `params.ID` is provided otherwise the update will fail at the incorrect URL.

Reply:
It seems I should add `ErrMissingDNSRecordID` or something like that, but should I put it in `dns.go` or the global `errors.go`?

Reply:
`dns.go` is fine. `errors.go` is used for library wide errors.

Reply:
Done. The same checking was added to `GetDNSRecord`.